__tests__/classes/MembershipExpirationProcessor.js

```diff
@@ -0,0 +1,154 @@
+// eslint-disable-next-line no-unused-vars
+/* global jest, describe, test, expect, jest, beforeEach, afterEach */
+
+const ah = require('../../test/ah-setup')
+const MembershipExpirationProcessor = require('../../classes/MembershipExpirationProcessor')
+const { generateMember, generatePayment } = require('../../test/generators')
+const addDays = require('date-fns/add_days')
+
+const expectedError = new Error('Provide time period!')
+const fromDate = new Date()
+
+describe('MembershipExpiration', () => {
+  const setup = () => {
+    return new MembershipExpirationProcessor({
+      api: ah.api,
+      config: {
+        daysBeforeExpiration: 30,
+        emailTemplateName: ''
+      }
+    })
+  }
+  const generateMemberAndPayment = async ({ memberOpts = { }, paymentOps = { } }) => {
+    const member = await ah.api.models.member.create(generateMember({
+      ...memberOpts
+    }))
+    await ah.api.models.payment.create(generatePayment({
+      paymentDate: new Date(),
+      paymentType: `cash`,
+      billingMemberId: member.id,
+      members: [member.id],
+      ...paymentOps
+    }))
+  }
+  beforeAll(async () => {
+    await ah.start()
+  })
+  afterAll(ah.stop)
+  beforeEach(async () => {
+    await ah.api.models.payment.destroy({ where: {}, force: true })
+    await ah.api.models.member.destroy({ where: {}, force: true })
+  })
+
+  describe('processing memberships for expiration', () => {
+    test('notify exipiring members with membershipEndDate in def interval', async () => {
+      const expirationProcessor = setup()
+      const sendMailSpy = jest.spyOn(expirationProcessor, 'enqueueSendMail')
+
+      await generateMemberAndPayment({ memberOpts: { membershipEndDate: addDays(fromDate, 27) } })
+      await generateMemberAndPayment({ memberOpts: { membershipEndDate: addDays(fromDate, 31) } })
+
+      await expirationProcessor.processMemberships(fromDate, addDays(fromDate, 30))
+
+      expect(sendMailSpy).toHaveBeenCalledTimes(1)
```

Comment:
```suggestion
      expect(sendMailSpy).toHaveBeenCalledWith([expect.objectContaining({id: expiringMember.id})])
```

This will make more explicit expectation - currently it's uncertain which one actually received email